contracts/DomainRegistry.sol

```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.20;
+
+contract DomainRegistry {
+    uint256 constant REQUIRED_DEPOSIT = 1 ether;
+
+    struct Domain {
+        uint256 deposit;
+        bool isRegistered;
+    }
+
+    mapping(string => Domain) public domains;
```

Comment:
Looking at this struct `Domain` and state var `domains` can't you see the problem? How can you answer which address in a domain controller?
-3 points

Reply:
fxd